superchain/superchain.go

```diff
@@ -99,10 +99,52 @@ type ChainConfig struct {
 	// This matches the resource filename, it is not encoded in the config file itself.
 	Chain string `yaml:"-"`
 
+	// UsePlasma is activated when the chain is in plasma mode.
+	UsePlasma    *bool        `json:"use_plasma,omitempty" yaml:"use_plasma,omitempty"`
+	Plasma       PlasmaConfig `json:"plasma,omitempty" yaml:"plasma,omitempty"`
+	PlasmaConfig `yaml:",inline"`
```

Comment:
Will this result in plasma config being duplicated? From what I can tell, after `CheckPlasma()` is called, we will have `nil` values embedded / inlined, and non-nil values nested under `Plasma`. I wonder if there is a better approach?

Reply:
[This](https://github.com/ethereum-optimism/superchain-registry/blob/5c3f1ca3313bfe61bc6c112d2fb6828171293795/add-chain/testdata/superchain/configs/sepolia/expected_plasma.yaml) is what the .yaml will look like.

The code that removes the inline fields is [here](https://github.com/ethereum-optimism/superchain-registry/blob/5c3f1ca3313bfe61bc6c112d2fb6828171293795/superchain/superchain.go#L138-L142). This works because these fields types are pointers and the yaml tag includes the `omitempty` directive

Reply:
I'm more concerned about the Go types exported from the `superchain` module. Can you confirm what they will look like?

Reply:
Good call. Ended up using a separate struct for Unmarshal to reduce downstream confusion risk: https://github.com/ethereum-optimism/superchain-registry/commit/c038f4c9529edb19c328c51ace862ec818d4ae3c